src/userService.ts

```diff
@@ -0,0 +1,134 @@
+import { users } from "./mockedData";
+import { Hobbies, User } from "./types";
+
+export class UserService {
+    private static instance: UserService | null = null;
+
+    private usersData: User[] = [];
+
+    private constructor() {
+        this.usersData = users;
+    }
+
+    public static getInstance(): UserService {
+        if (!UserService.instance) {
+            UserService.instance = new UserService();
+        }
+        return UserService.instance;
+    }
+
+    getAllUsers(): User[] {
+        return this.usersData;
+    };
+
+    getUserById(id: number): Partial<Omit<User, 'hobbies'>> | null {
+        if (id) {
+            const user = this.usersData.find(user => user.id === id) as Partial<User> || null;
+            const reternedUser = {...user};
+            if (user) {
+                delete reternedUser.hobbies
+            }
+            return reternedUser
+        }
+        return null
+        
+    };
+
+    getUserHobbies(id: number): Hobbies | null {
+        if (id) {
+            const user = this.usersData.find(user => user.id === id);
+            
+            return user?.hobbies || null
+        }
+        
+        return null
+    };
+
+    createUser(user: Omit<User, 'id'>): User | null {
+        if (user) {
+            const newUser: User = {
+                ...user,
+                id: Date.now()
+            }
+            this.usersData.push(newUser)
+
+            return newUser
+        } 
+
+        return null
+    }
+
+    deleteUser(id: number): User[] | null {
+        if (id) {
+            return this.usersData = this.usersData.filter(user => user.id !== id)
+        }
+
+        return null
+    }
+
+    updateUser(id: number, partialUser: Partial<Omit<User, "id">>): User | null {
+        if (id && partialUser) {
+            let newUser: unknown;
+            this.usersData = this.usersData.map(user => {
+                if (user.id === id) {
+                    newUser = {
+                        ...user,
+                       ...partialUser
+                    }                    
+                    return newUser as User
+                }
+                return user
+            })
+
+            return newUser as User
+        }
+
+        return null
+    }
```

Comment:
I would suggest to rewrite it like this

```suggestion
    updateUser(id: number, partialUser: Partial<Omit<User, "id">>): User | null {
        if (id && partialUser) {
            const index = this.usersData.findIndex(({ id: userId }) => userId === id);

            if(index !== -1) {
                const currentUser = this.usersData[index];
                const updatedUser = {
                   ...currentUser,
                   ...partialUser
                };
                
                this.usersData[index] = updatedUser;
                return updatedUser;
            }
        }

        return null
    }
```